e2e/rbd.go

```diff
@@ -545,6 +547,402 @@ var _ = Describe("RBD", func() {
+				}
+				// Storage class with rbd-nbd mounter
+				err = createRBDStorageClass(
+					f.ClientSet,
+					f,
+					defaultSCName,
+					nil,
+					map[string]string{
+						"mounter":         "rbd-nbd",
+						"mapOptions":      nbdMapOptions,
+						"cephLogStrategy": e2eDefaultCephLogStrategy,
+					},
+					deletePolicy)
+				if err != nil {
+					e2elog.Failf("failed to create storageclass: %v", err)
+				}
+				pvc, err := loadPVC(rawPvcPath)
+				if err != nil {
+					e2elog.Failf("failed to load PVC: %v", err)
+				}
+				pvc.Namespace = f.UniqueName
+				pvc.Spec.AccessModes = []v1.PersistentVolumeAccessMode{v1.ReadWriteMany}
+
+				app, err := loadAppDeployment(rxRawAppPath)
+				if err != nil {
+					e2elog.Failf("failed to load application deployment: %v", err)
+				}
+				app.Namespace = f.UniqueName
+
+				err = createPVCAndDeploymentApp(f, "", pvc, app, deployTimeout)
+				if err != nil {
+					e2elog.Failf("failed to create PVC and application: %v", err)
+				}
+
+				err = waitForDeploymentComplete(app.Name, app.Namespace, f.ClientSet, deployTimeout)
+				if err != nil {
+					e2elog.Failf("timeout waiting for deployment to be in running state: %v", err)
+				}
+
+				devPath := app.Spec.Template.Spec.Containers[0].VolumeDevices[0].DevicePath
+				cmd := fmt.Sprintf("dd if=/dev/zero of=%s bs=1M count=10", devPath)
+
+				opt := metav1.ListOptions{
+					LabelSelector: fmt.Sprintf("app=%s", app.Name),
+				}
+				podList, err := f.PodClientNS(app.Namespace).List(context.TODO(), opt)
+				if err != nil {
+					e2elog.Failf("get pod list failed: %v", err)
+				}
+				if len(podList.Items) != int(*app.Spec.Replicas) {
+					e2elog.Failf("podlist contains %d items, expected %d items", len(podList.Items), *app.Spec.Replicas)
+				}
+				for _, pod := range podList.Items {
+					_, _, err = execCommandInPodWithName(f, cmd, pod.Name, pod.Spec.Containers[0].Name, app.Namespace)
+					if err != nil {
+						e2elog.Failf("command %q failed: %v", cmd, err)
+					}
+				}
+
+				err = deletePVCAndDeploymentApp(f, "", pvc, app)
+				if err != nil {
+					e2elog.Failf("failed to delete PVC and application: %v", err)
+				}
+				// validate created backend rbd images
+				validateRBDImageCount(f, 0, defaultRBDPool)
+				// validate images in trash
+				err = waitToRemoveImagesFromTrash(f, defaultRBDPool, deployTimeout)
+				if err != nil {
+					e2elog.Failf("failed to validate rbd images in pool %s trash with error %v", rbdOptions(defaultRBDPool), err)
+				}
+				err = deleteResource(rbdExamplePath + "storageclass.yaml")
+				if err != nil {
+					e2elog.Failf("failed to delete storageclass: %v", err)
+				}
+				err = createRBDStorageClass(f.ClientSet, f, defaultSCName, nil, nil, deletePolicy)
+				if err != nil {
+					e2elog.Failf("failed to create storageclass: %v", err)
+				}
+			})
+
+			By("Create ROX+FS Mode PVC and bind to multiple pods via deployment using rbd-nbd mounter", func() {
```

Comment:
@Madhu-1 I was following "create ROX PVC clone and mount it to multiple pods" test.

Reply:
just trying to keep this case simple and also reduce the testing time. we can update the existing test cases so that it will also reduce the testing time also

Reply:
:+1: